Nama : Dicki Pramana Putra  
NIM : 250401020112  
Kelas : IF-405  

In [ ]:
import pandas as pd
import numpy as np

# Muat dataset Titanic dari URL
url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"
df = pd.read_csv(url)

# Tampilkan informasi awal
print("Dataset berhasil dimuat!")
print(df.head())

Dataset berhasil dimuat!
   PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8

In [ ]:
# Q1: Berapa jumlah baris dan kolom?
print(f"Baris: {df.shape[0]}, Kolom: {df.shape[1]}")
print(f"Nama kolom: {df.columns.tolist()}")

Baris: 891, Kolom: 12
Nama kolom: ['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


In [ ]:
# Q2: Missing values per kolom
print("Jumlah missing values:")
print(df.isnull().sum())
print("\nProporsi missing (%):")
print((df.isnull().sum() / len(df) * 100).round(2))

Jumlah missing values:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Proporsi missing (%):
PassengerId     0.00
Survived        0.00
Pclass          0.00
Name            0.00
Sex             0.00
Age            19.87
SibSp           0.00
Parch           0.00
Ticket          0.00
Fare            0.00
Cabin          77.10
Embarked        0.22
dtype: float64


**Apa yang dipelajari:**
Pertemuan ini mempelajari eksplorasi data (EDA) menggunakan Pandas pada dataset Titanic
yang berisi 891 baris dan 12 kolom, termasuk filtering, groupby, dan agregasi data.

**Temuan utama:**
Sebanyak 38.38% penumpang selamat. Penumpang wanita kelas 1 memiliki tingkat keselamatan
tertinggi (91 dari 94 orang selamat). Kelas 1 memiliki tingkat keselamatan 63%,
jauh lebih tinggi dibanding kelas 3 yang hanya 24.2%.